radix-engine/src/system/system_callback.rs

```diff
@@ -1571,7 +1571,61 @@ impl<V: SystemCallbackObject> KernelTransactionExecutor for System<V> {
                         )
                     }
                 }
-            };
+            }
+            .and_then(|_| {
+                match hash_nullification {
+                    IntentHashNullification::TransactionIntent { .. } => {
+                        // Transaction intent nullification is historically not costed.
+                    }
+                    IntentHashNullification::Subintent { .. } => {
+                        if let Some(costing) = modules.costing_mut() {
+                            let read_not_found =
+                                IOAccess::ReadFromDbNotFound(CanonicalSubstateKey {
+                                    node_id: TRANSACTION_TRACKER.into_node_id(),
+                                    partition_number: PartitionNumber(0),
```

Comment:
This appears a little hacky (given that I believe the original intention of `IOAccess` was to be able to track exactly what was read).

Could we perhaps either:
* Use the actual substate key? Say via `validate_intent_hash_uncosted` returning it?
* Acknowledge the hack with a commented line `let placeholder_key = CanonicalSubstateKey { ... }`, and we can then reuse `placeholder_key` in the two places below.

Reply:
Agree with the sentiment, although ultimately this probably should be replaced with a constant (with the current logic as a basis), which is why I didn't bother making it neater.

Will try to clean this up a little bit.